src/components/Incidents/IncidentsPage.jsx

```diff
@@ -10,67 +10,62 @@ import { useDispatch } from 'react-redux';
+  { endTime: 1724748294404, duration: 86400000 },
+];
 
 const IncidentsPage = ({
   customDataSource,
   defaultTimespan = parsePrometheusDuration('7d'),
-  defaultSamples,
   namespace,
   timespan,
-  fixedXDomain,
 }) => {
   const [startingDate, setStartingDate] = React.useState('');
   const [endDate, setEndDate] = React.useState('');
   const [incidentsPageData, setIncidentsPageData] = React.useState([]);
-  const defaultSpanText = spans.find((s) => parsePrometheusDuration(s) >= defaultTimespan);
-  const [span, setSpan] = React.useState(timespan || parsePrometheusDuration(defaultSpanText));
   //raw data
   const [data, setData] = React.useState([]);
   //data that is mapped and changed from timestamps to a format HH/DD/MM/YY
   const [processedData, setProcessedData] = React.useState([]);
   //will be used to define the Xdomain of chart
-  const [xDomain, setXDomain] = React.useState();
-  const maxSamplesForSpan = defaultSamples || getMaxSamplesForSpan(span);
-
-  const [samples, setSamples] = React.useState(maxSamplesForSpan);
-  const getXDomain = (endTime, span) => [endTime - span, endTime];
-  React.useEffect(() => {
-    setXDomain(getXDomain(Date.now(), span));
-  }, [span, fixedXDomain]);
-  const endTime = xDomain?.[1];
-
-  React.useEffect(() => {
-    if (timespan) {
-      setSpan(timespan);
-      setSamples(defaultSamples || getMaxSamplesForSpan(timespan));
-    }
-  }, [defaultSamples, timespan]);
+  const [xDomain, setXDomain] = React.useState(twoWeeksDateRanges);
   const safeFetch = useSafeFetch();
-  const now = Date.now();
-  const timeRanges = getTimeRanges(span, endTime || now);
-
   React.useEffect(() => {
     (async () => {
       try {
-        const response = await safeFetch(
-          getPrometheusURL(
-            {
-              endpoint: PrometheusEndpoint.QUERY_RANGE,
-              endTime: 1722950372519,
-              namespace,
-              query: 'ALERTS',
-              samples: Math.ceil(samples / timeRanges.length),
-              timespan: 86400000 - 1,
-            },
-            customDataSource?.basePath,
-          ),
+        const results = await Promise.all(
+          twoWeeksDateRanges.map(async (range) => {
+            const response = await safeFetch(
+              getPrometheusURL(
+                {
+                  endpoint: PrometheusEndpoint.QUERY_RANGE,
+                  endTime: range.endTime,
+                  namespace,
+                  query: 'ALERTS',
+                  samples: 23,
+                  timespan: range.duration - 1,
+                },
+                customDataSource?.basePath,
+              ),
+            );
+            return response.data.result;
+          }),
         );
-        setData(response.data.result);
+        const aggregatedData = results.reduce((acc, result) => acc.concat(result), []);
+        setData(aggregatedData);
       } catch (err) {
         // eslint-disable-next-line no-console
         console.log(err);
```

Comment:
When working with promises, you can use `.then()`, `.catch()`, and `.finally()` on the promise itself to help handle the results of it rather than keeping track of scopes of try catch blocks and awaits. It is mostly a code formatting preference, but that is how we tend to have our code in our plugins

```suggestion
        Promise.all(
          twoWeeksDateRanges.map(async (range) => {
            const response = await safeFetch(
              getPrometheusURL(
                {
                  endpoint: PrometheusEndpoint.QUERY_RANGE,
                  endTime: range.endTime,
                  namespace,
                  query: 'ALERTS',
                  samples: 23,
                  timespan: range.duration - 1,
                },
                customDataSource?.basePath,
              ),
            );
            return response.data.result;
          }),
        ).then((results) => {
          const aggregatedData = results.reduce((acc, result) => acc.concat(result), []);
          setData(aggregatedData);
        })
        .catch((err) => {
          // eslint-disable-next-line no-console
          console.log(err);
        })
```